src/Symfony/Component/Serializer/Tests/Debug/TraceableSerializerTest.php

```diff
@@ -126,6 +126,40 @@ public function testAddDebugTraceIdInContext()
         $traceableSerializer->encode('data', 'format');
         $traceableSerializer->decode('data', 'format');
     }
+
+    public function testCollectedCaller()
+    {
+        $serializer = new \Symfony\Component\Serializer\Serializer();
+
+        $collector = new SerializerDataCollector();
+        $traceableSerializer = new TraceableSerializer($serializer, $collector);
+
+        $traceableSerializer->normalize('data');
+        $collector->lateCollect();
+
+        $this->assertSame([
+            'name' => 'TraceableSerializerTest.php',
+            'file' => __FILE__,
+            'line' => __LINE__ - 6,
+        ], $collector->getData()['normalize'][0]['caller']);
+    }
+
+    public function testCollectedCallerFromArrayMap()
+    {
+        $serializer = new \Symfony\Component\Serializer\Serializer();
+
+        $collector = new SerializerDataCollector();
+        $traceableSerializer = new TraceableSerializer($serializer, $collector);
+
+        array_map([$traceableSerializer, 'normalize'], ['data']);
```

Comment:
When using `array_map` with a callable like above or a native first class callable like `$traceableSerializer->normalize(...)` the problem occurs. Somehow PHP does not provide a `file` and `line` for that location when using [debug_backtrace](https://www.php.net/manual/en/function.debug-backtrace.php).

![Screenshot 2025-06-17 at 12 15 54@2x](https://github.com/user-attachments/assets/950f1b94-db3e-4117-be4b-3583972772e2)